### PR TITLE
First batch for #665

### PR DESCRIPTION
**Pull request by Samueru-sama**

### CODE REVIEW

**Comment by ivan-hc** [comment]
Creating installation scripts for portable apps in TAR/ZIP is a long and exhausting procedure. A workflow run for github would be needed to verify the contents of all the files contained in an archive, so as to avoid having to download everything and verify directly.

**Comment by ivan-hc** [comment]
Something like this:
```
name: Test archive content
concurrency:
  group: build-${{ github.ref }}
  cancel-in-progress: true

on:
  workflow_dispatch:

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3

    - name: build
      if: always()
      run: |
        wget https://github.com/user/repo/releases/download/$tag/archive-x86_64.tar
        tar xf ./*.tar
        ls .
```

**Comment by Samueru-sama** [comment]
>Creating installation scripts for portable apps in TAR/ZIP is a long and exhausting procedure.

That isn't much of a problem to me, because I have to test the script anyway. (And I have fast internet) the issue is when the file is on a random website that's hard to parse.

**Comment by ivan-hc** [comment]
Just ask me if you have issues, do not work completelly alone.

Also, I'm writing the commands in the issue hoping someone will learn how to write the commands too, for educational purposes.

**Comment by Samueru-sama** [comment]
I added mullvad but while the script, updater and everything works. 

The app itself doesn't work on my PC lol. Nothing happens when I launch their start script that they ship in their browser lol

**Comment by ivan-hc** [comment]
> The app itself doesn't work on my PC. Nothing happens when I launch their start script that they ship in their browser lol

it's their problem. If users find their apps, especially through our package manager, they just need to run `am -a` and they will also have the URL to the official site or repository or any source, where they can report the bug (that's why I need the .md file, if github supports uploading .zip archives, I just need to upload the `am-scripts/portable-linux-apps.github.io/apps` folder, I will make the necessary corrections)

**Comment by Samueru-sama** [comment]
> Just ask me if you have issues, do not work completelly alone.
> 
> Also, I'm writing the commands in the issue hoping someone will learn how to write the commands too, for educational purposes.

I'll make some dinner and continue later in the night btw

**Comment by ivan-hc** [comment]
In the meantime we have reached the 2040 apps quota in "main" before we merge this PR... 

Istantanea_2024-06-16_05-34-46 png

...so now I can update the description of the repository.

**Comment by Samueru-sama** [comment]
I added multimc, because it has a fixed download url without version I did a  hack in the `AM-updater`, take a look and let me know what you think

> In the meantime we have reached the 2040 apps quota in "main" before we merge this PR...

Btw I think yuzu needs to be removed or wrapped to point to suyu instead. The project has been dead for months due to a lawsuit from nintendo.

**Comment by ivan-hc** [comment]
> I added multimc, because it has a fixed download url without version I did a hack in the `AM-updater`, take a look and let me know what you think

I think this is the best we can do, I've seen their article on their site

https://multimc.org/posts/0-7-0-update.html

it seems that they will no more use the "release" concept... is this app autoupdatable?

If so, we can also remove the AM-updater completelly.

PS: don't forget to add the entry on the list.

About the .md files, I can keep create them by myself, but I need all the entries on the list.

> Btw I think yuzu needs to be removed or wrapped to point to suyu instead. The project has been dead for months due to a lawsuit from nintendo.

If the repo has not be deleted, we can still distribute it, also if unsupported. Maybe someone needs it. Its worth it to keep all the AppImages still downloadable and (more or less) updatable (also if in archived repositories).

It is the user who must trust the app and the developer, we are not the ones that must apply censorship.

**Comment by ivan-hc** [comment]
> It is the user who must trust the app and the developer, we are not the ones that must apply censorship.

For example, after all I've seen about Thorium browser on YouTube, after Chris Thitus review... I'll never trust that project. However, there are still people that use it, and someone that have suggested many other packages from the same developer, Mecury browser included... and all their variants.

That said, if there is still people that trust app and developer, we are none to prevent them to have/use these apps.

**Comment by Samueru-sama** [comment]
> About the .md files, I can keep create them by myself, but I need all the entries on the list.

I don't have md files besides the one for the only appimage I made, since I made most of these manually by copying another script. 

All the recent CLI applications that I added use a modified version of the AppImage template that I haven't had time to add to am sorry. 
[mcpelauncher.md](https://github.com/user-attachments/files/15856111/mcpelauncher.md)

**Comment by ivan-hc** [comment]
> EDIT: Btw what I mean with yuzu is that download link is dead and will be forever, it's gone.
> 
> I think it can be switched to a unofficial mirror if you want.

AH! So the voice must be removed from the list... and even the script.

**Comment by ivan-hc** [comment]
Just tested the URL, its dead... and also the one on the AUR is dead https://aur.archlinux.org/packages/yuzu-appimage

The only two available are at https://archive.org/download/citra-qt-and-yuzu-EA but... if its a so illegal app to be available only on the web archive... we should remove it.

I'll wait you finish to compile this PR and then we will remove yuzo.

**Comment by ivan-hc** [comment]
You forgot to add "multimc" to the list, so I did it for you.

**Comment by Samueru-sama** [comment]
> You forgot to add "multimc" to the list, so I did it for you.

I was going to do it, I was making the badlion script. Thank you.

**Comment by ivan-hc** [comment]
I've also opened a PR to keep your repo in sync with main

**Comment by Samueru-sama** [comment]
Btw badlion also has the hack that the `AM-updater` always updates the app. 

Is this also a self updatetable application?

**Comment by Samueru-sama** [comment]
I'm stuck with modrinth as the appimage download link is actually versioned this time.

**Comment by ivan-hc** [comment]
> Btw badlion also has the hack that the `AM-updater` always updates the app.
> 
> Is this also a self updatetable application?

I gave you a command to intercept the URL available for the NixOS package manager. But perhaps we should do differently:
```
SITE="https://raw.githubusercontent.com/NixOS/nixpkgs/master/pkgs/games/badlion-client/default.nix"
version=$(wget -q "$SITE" -O - | grep -i 'version' | tr '"' '\n' | grep [0-9]  | head -1)
wget "$(wget -q "$SITE" -O - | sed 's/[()",{}]/ /g; s/ /\n/g' | grep -oi 'https.*badlion.*client$' | head -1)" || exit 1
```
so we can get the version number to compare the versions and download the new package with wget

**Comment by ivan-hc** [comment]
When on a page, you see a linear URL, I recommend writing the entire command for the linear URL (maybe it will change one day?) for `wget`, while the $version variable must be a command that intercepts a version, a date... a unique number. That's enough to have a comparison.

**Comment by Samueru-sama** [comment]
modrinth has forbidden wget requests 🤮

**Comment by ivan-hc** [comment]
try this
```
wget -q https://modrinth.com/app -O - | tr '"' '\n' | grep http | grep -i appimage$ | head -1
```

**Comment by ivan-hc** [comment]
Istantanea_2024-06-16_07-29-55 png

**Comment by ivan-hc** [comment]
```
ivan@debian:~$ am -t modrinth
##############################################################

 CREATE TEMPLATES FOR "MODRINTH"
				
--------------------------------------------------------------

 EACH MODEL IS BASED ON A DIFFERENT METHOD OF BUILDING/UPDATING THE PROGRAM.
				
 PLEASE, SELECT A TEMPLATE FOR "MODRINTH": 
				
  0) APPIMAGE, FROM ANY WEBSITE (BETTER IF FROM GITHUB.COM)
  1) APPIMAGE, BUILD YOUR OWN WITH PKG2APPIMAGE
  2) ANY ARCHIVE/BINARY/SCRIPT FROM ANY WEBSITE
  3) WEBAPP BASED ON A FIREFOX PROFILE (REQUIRES "FIREFOX" IN $PATH)"
 
 WHAT KIND OF PROGRAM DO YOU WANT TO WRITE A SCRIPT FOR? : 0
-----------------------------------------------------------------------
 ◆ ADD THE URL OF THE SITE (BETTER IF IT COME FROM GITHUB), FOR EXAMPLE: 

 - if from github.com, use "https://github.com/USER/PROJECT"

 - if from other sources, use "https://www.SITENAME.com" or something

 URL: https://modrinth.com
 ◆ ADD A BRIEF DESCRIPTION FOR THIS APPLICATION 
 : The Modrinth App is a unique, open source launcher that allows you to play your favorite mods, and keep them up to date, all in one neat little package.
-----------------------------------------------------------------------
 USE A ONE-LINE SHELL COMMAND TO CHECK ONLY THE URL TO THE APPIMAGE 

 if the URL is fixed, simply add the "echo" command at the beginning

 : wget -q https://modrinth.com/app -O - | tr '"' '\n' | grep http | grep -i appimage$ | head -1
-----------------------------------------------------------------------

 All files are saved in /home/ivan/Scrivania/am-scripts

ivan@debian:~$ appman -i '/home/ivan/Scrivania/am-scripts/x86_64/modrinth'
```

**Comment by ivan-hc** [comment]
Well, one left?

**Comment by Samueru-sama** [comment]
Yeah I use the link you gave, thank you, let me add the description

**Comment by ivan-hc** [comment]
I must say that you gave me the idea of use `tr '"' '\n'`, and all times I test a URL I use both wget and wget2

**Comment by Samueru-sama** [comment]
> I must say that you gave me the idea of use `tr '"' '\n'`, and all times I test a URL I use both wget and wget2

I normally use the `sed 's/[()",{}]/ /g; s/ /\n/g'` instead as it applies the trick to most of the json separators. 

The trick also doesn't really need the `jq` before it, but I leave it just because.

If all scripts use this jq can be dropped as dependency btw. Specially because I noticed that it explodes on sites like librewolf lol.

**Comment by ivan-hc** [comment]
```
ivan@debian:~/Scrivania$ wget2 -q https://api.github.com/repos/ivan-hc/Anydesk-appimage/releases -O - | tr '"' '\n' | grep -i appimage$ | cut -d '"' -f 4
Anydesk-6.3.2-x86_64.AppImage
https://github.com/ivan-hc/Anydesk-appimage/releases/download/continuous/Anydesk-6.3.2-x86_64.AppImage
ivan@debian:~/Scrivania$ wget -q https://api.github.com/repos/ivan-hc/Anydesk-appimage/releases -O - | tr '"' '\n' | grep -i appimage$ | cut -d '"' -f 4
Anydesk-6.3.2-x86_64.AppImage
https://github.com/ivan-hc/Anydesk-appimage/releases/download/continuous/Anydesk-6.3.2-x86_64.AppImage
```
no `jq` needed

**Comment by ivan-hc** [comment]
```
ivan@debian:~/Scrivania$ wget -q https://api.github.com/repos/ivan-hc/Anydesk-appimage/releases -O - | tr '"' '\n' | grep "^http" |  grep -i appimage$ | cut -d '"' -f 4
https://github.com/ivan-hc/Anydesk-appimage/releases/download/continuous/Anydesk-6.3.2-x86_64.AppImage
ivan@debian:~/Scrivania$ wget2 -q https://api.github.com/repos/ivan-hc/Anydesk-appimage/releases -O - | tr '"' '\n' | grep "^http" |  grep -i appimage$ | cut -d '"' -f 4
https://github.com/ivan-hc/Anydesk-appimage/releases/download/continuous/Anydesk-6.3.2-x86_64.AppImage
```
F*ck jq!

**Comment by Samueru-sama** [comment]
> no `jq` needed

Indeed, the trick is not depending on whitespace when grepping. Which the sed/tr fixes by turning the `"` into new lines first. 

This is also why I normally use one grep only with wildcards. and I add a `$` at the end, to indicate that I want a link that ends in the last character of the grep. that way you don't have to `grep -v zsync` for example

**Comment by ivan-hc** [comment]
Hombre, desearía haber sido más consciente entonces. Habría evitado agregar esta dependencia innecesaria.

Let me merge this dependence.

**Comment by Samueru-sama** [comment]
> Hombre, desearía haber sido más consciente entonces. Habría evitado agregar esta dependencia innecesaria.

My italian isn't the best, but I mentioned this before? I was the one that updated the `INSTALL` script to use that very trick.

**Comment by ivan-hc** [comment]
Damn, it's 8 in the morning!

I go create md files to update the site and relax a bit. I'm a vampire now, the way I live.

Thanks again.

**Comment by Samueru-sama** [comment]
Btw I have a irrational hatred with jq, too complicated to use for me. 

I migrated all my personal scripts that parse json to use `gron.awk` which is a very simple awk scripts that works in any POSIX compliant PC. It even works on android. 

But for am, since we normally only need urls, we can really get away without using any such tool and only use the sed/tr trick. But that means all 2000 scripts have to be updated to such trick. And I don't that even 100 have been updated.

**Comment by ivan-hc** [comment]
All of them have been updated https://github.com/ivan-hc/AM/commit/286f936c63e9359661aa84dc44b0d58ad5c62174

**Comment by Samueru-sama** [comment]
> All of them have been updated [286f936](https://github.com/ivan-hc/AM/commit/286f936c63e9359661aa84dc44b0d58ad5c62174)

No I mean to remove jq as a dependency, the greps for `version` have to have the sed/tr trick first, and very few actually do. 

What `jq` is doing is making sure that the json is human readable as that is how those greps were being used with. When the json is compact everything turns into a single line, and that breaks greps that depended on new lines / whites spaces as separators. 

jq was actually the only quick solution available that works with existing code.
